app/models/api/openid_connect/authorization.rb

```diff
@@ -0,0 +1,62 @@
+module Api
+  module OpenidConnect
+    class Authorization < ActiveRecord::Base
+      belongs_to :user
+      belongs_to :o_auth_application
+
+      validates :user, presence: true
+      validates :o_auth_application, presence: true
+      validates :user, uniqueness: {scope: :o_auth_application}
+
+      has_many :authorization_scopes
+      has_many :scopes, through: :authorization_scopes
+      has_many :o_auth_access_tokens, dependent: :destroy
+      has_many :id_tokens, dependent: :destroy
+
+      before_validation :setup, on: :create
+
+      scope :with_redirect_uri, ->(given_uri) { where redirect_uri: given_uri }
+
+      def setup
+        self.refresh_token = SecureRandom.hex(32)
+      end
+
+      def accessible?(required_scopes=nil)
+        Array(required_scopes).all? do |required_scope|
+          scopes.include? required_scope
+        end
+      end
+
+      def create_code
+        self.code = SecureRandom.hex(32)
+        save
+        code
+      end
+
+      def create_access_token
+        o_auth_access_tokens.create!.bearer_token
+        # TODO: Add support for request object
```

Comment:
Yeah this is for the "Implement request objects, including [require and require_uri](http://openid.net/specs/openid-connect-core-1_0.html#JWTRequests)". I've noted it down under "Remaining Tasks for Future API Release" as it is optional under OIDC; I'll delete this TODO for now.